src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php

```diff
@@ -35,8 +36,8 @@ class XmlEncoderTest extends TestCase
     protected function setUp(): void
     {
         $this->encoder = new XmlEncoder();
-        $serializer = new Serializer([new CustomNormalizer()], ['xml' => new XmlEncoder()]);
-        $this->encoder->setSerializer($serializer);
+        $normalizer = new ChainNormalizer([new CustomNormalizer()]);
+        $serializer = new Serializer([], ['xml' => $this->encoder], $normalizer);
```

Comment:
```suggestion
        $serializer = new Serializer([], ['xml' => $this->encoder], new CustomNormalizer());
```
Same for other occurrences